src/asset-relocator.js

```diff
@@ -108,7 +109,9 @@ function getEntryIds (compilation) {
     }
     else if (typeof compilation.options.entry === 'object') {
       try {
-        return Object.values(compilation.options.entry).map(entry => resolve.sync(entry, { extensions }));
+        return flattenArray(Object.values(compilation.options.entry)
+          .map(entry => typeof entry === "string" ? [entry] : flattenArray(Object.values(entry)))
+        ).map(entryString => resolve.sync(entryString, { extensions }));
```

Comment:
If the `"import"` form is the specific pattern I would rather match the `"import"` form exactly rather than doing a generic flattening.

Webpack seems to change these APIs quite a bit, so trying to do conservative matching might be worthwhile to avoid future breaks too.

Reply:
You're right
https://github.com/webpack/webpack/blob/0401f6bff138346eaf4a7fb380f819943b24ad1e/types.d.ts#L1957-L1977

It indeed seems the case, I'll refactor that.